### DISCUSSION TITLE
What level of CPU usage is to be expected?

### DISCUSSION THREAD

**Discussion by jlaitinen81** [blakeblackshear/frigate#714] [Ask A Question]
I know that there are a number of variables, but I was wondering if a CPU load increase from 6% to 23% is reasonable on a Home Assistant (HassOS) dedicated 5th gen NUC (Intel® Core™ i7-5557U Processor, 12GB RAM) when enabling the Frigate Addon? Im running 5 cameras, each with 5FPS detection and native resolution clips. I have a Coral USB dongle installed.

My ffmpeg settings are as follows:
![image](https://user-images.githubusercontent.com/67101282/106945632-b5bbb180-6730-11eb-8127-fe2b5501feca.jpeg)

**Comment by jlaitinen81** [comment]
As a follow up I wanted to mention that my cameras do not support lower quality substreams and I have just set the detection to use native resolution with 5FPS even if the streams are actually higher FPS.

I think I might be stressing the CPU with continuous stream processing, but I don't know how to solve this...

**Comment by paularmstrong** [comment]
That increase seems to align with my setup. I have 3 cameras running on a 10th gen NUC with lower resolution substreams. Each camera requires at least one ffmpeg process while there is no motion and each ffmpeg process takes 3-5% cpu. Add your 5 together and you're easily accounting for a 20% increase.

**Reply by jlaitinen81** [reply]
Thanks Paul! If your 10th Gen NUC uses 3-5% CPU capacity per camera then I think there is no reason for me to consider upgrading my NUC.

Wonder if there is any way to reduce the CPU load meaningfully. Would like to bring it to 10-15% all in all, as that would allow the CPU to run cooler and passive cooling to be enough.

**Reply by blakeblackshear** [reply]
Most of the CPU usage comes from ffmpeg decoding. If you are already using hwaccel, the only way to drop it further is to use a lower resolution and/or ensure you have limited the frame rate on the camera itself. It just takes a good bit of processing to decode video feeds.
